elrond-wasm/src/io/dyn_arg_input_managed_result.rs

```diff
@@ -0,0 +1,60 @@
+use crate::{
+    api::{ErrorApi, ManagedTypeApi},
+    err_msg,
+    types::{ManagedBuffer, ManagedVec},
+    DynArgInput,
+};
+
+pub struct ManagedResultArgLoader<A>
+where
+    A: ManagedTypeApi + ErrorApi,
+{
+    data: ManagedVec<A, ManagedBuffer<A>>,
+    data_len: usize,
+    next_index: usize,
+    api: A,
+}
+
+impl<A> ManagedResultArgLoader<A>
+where
+    A: ManagedTypeApi + ErrorApi,
+{
+    pub fn new(api: A, data: ManagedVec<A, ManagedBuffer<A>>) -> Self {
+        let data_len = data.len();
+        let mut result = ManagedResultArgLoader {
+            data,
+            data_len,
+            next_index: 0,
+            api,
+        };
+        result
```

Comment:
Clippy complaint: Return the `MaangedResultArgLoader` directly, no `let` needed.